client/signup/main.jsx

```diff
@@ -227,21 +215,21 @@ class Signup extends React.Component {
 		}
 	}
 
-	updateLoadingScreenStartTime = ( progress = this.props.progress ) => {
+	updateShouldShowLoadingScreen = ( progress = this.props.progress ) => {
 		const hasInvalidSteps = !! getFirstInvalidStep( this.props.flowName, progress ),
 			waitingForServer = ! hasInvalidSteps && this.isEveryStepSubmitted( progress ),
-			startLoadingScreen = waitingForServer && ! this.state.loadingScreenStartTime;
+			startLoadingScreen = waitingForServer && ! this.state.shouldShowLoadingScreen;
 
-		if ( this.isEveryStepSubmitted( progress ) ) {
+		if ( ! this.isEveryStepSubmitted( progress ) ) {
```

Comment:
This is a janitorial fix. Even though the boolean logic was wrong, it does not cause issues since `goToFirstInvalidStep ()` bails out if there are no invalid steps. Nevertheless, it's good to have the correct logic here 😄